apps/voting/app/src/components/VotePanelContent.js

```diff
@@ -180,30 +205,82 @@ class VotePanelContent extends React.Component {
           ready={ready}
         />
 
-        {userCanVote && (
-          <div>
-            <SidePanelSeparator />
-            <VotingButtons>
-              <Button
-                mode="strong"
-                emphasis="positive"
-                wide
-                onClick={this.handleYesClick}
-              >
-                Yes
-              </Button>
-              <Button
-                mode="strong"
-                emphasis="negative"
-                wide
-                onClick={this.handleNoClick}
-              >
-                No
-              </Button>
-            </VotingButtons>
-            <Info title={`You will cast ${userBalance || '...'} votes`} />
-          </div>
-        )}
+        {(() => {
+          if (canExecute) {
+            return (
+              <div>
+                <SidePanelSeparator />
+                <ButtonsContainer>
+                  <Button mode="strong" wide onClick={this.handleExecuteClick}>
+                    Execute vote
+                  </Button>
+                </ButtonsContainer>
+                <Info.Action>
+                  Executing this vote is required to enact it.
+                </Info.Action>
+              </div>
+            )
+          }
+
+          if (userCanVote && hasVoted && !changeVote) {
+            return (
+              <div>
+                <SidePanelSeparator />
+                <ButtonsContainer>
+                  <Button
+                    mode="strong"
+                    wide
+                    onClick={this.handleChangeVoteClick}
+                  >
+                    Change my vote
+                  </Button>
+                </ButtonsContainer>
+                <Info.Action>
+                  You voted yes with {userBalance || '…'} tokens.
+                </Info.Action>
+              </div>
+            )
+          }
+
+          if (userCanVote) {
+            return (
+              <div>
+                <SidePanelSeparator />
+                <ButtonsContainer>
+                  <VotingButton
+                    mode="strong"
+                    emphasis="positive"
+                    wide
+                    onClick={this.handleYesClick}
+                  >
+                    Yes
+                  </VotingButton>
+                  <VotingButton
+                    mode="strong"
+                    emphasis="negative"
+                    wide
+                    onClick={this.handleNoClick}
+                  >
+                    No
+                  </VotingButton>
+                </ButtonsContainer>
+                <Info.Action>
+                  <p>
+                    You will cast {userBalance || '…'} votes, since it was your
+                    token balance at the{' '}
+                    <SafeLink
+                      href={`${etherscanBaseUrl}/block/${snapshotBlock}`}
+                      target="_blank"
+                    >
+                      beginning of the vote
```

Comment:
I think we should print the block number here or add it on top right next to `quorum`, thoughts?

Reply:
Yes maybe we could have it at the top? @jounih 

In the meantime, I added it here.